frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -475,6 +482,19 @@ function ReaderPaging:onZoomModeUpdate(new_mode)
     self.zoom_mode = new_mode
 end
 
+function ReaderPaging:onZoomPanUpdate(settings)
+    local arrayContains = util.arrayContains
+    for k, v in pairs(settings) do
+        if k ~= "zoom_factor" then
+            if arrayContains(ReaderZooming.pan_zoom_settings, k) then
```

Comment:
Is the local copy necessary?
```suggestion
    for k, v in pairs(settings) do
        if k ~= "zoom_factor" then
            if util.arrayContains(ReaderZooming.pan_zoom_settings, k) then
```

Reply:
In terms of optimization, it's slightly better (avoids a table lookup within the loop). That said, I don't know how luajit optimizes that, and won't fight for it…

Reply:
It's... mostly irrelevant to counter-productive in LuaJIT, AFAIK. (c.f., the recommendation to cache the C namespace vs. individual functions for FFI stuff).